lib/modules/wdpa/s3.rb

```diff
@@ -47,7 +52,9 @@ def current_wdpa
   end
```

Comment:
Also, I'm a little bit concerned that `current_wdpa` is still being referenced across the codebase but that `current_wdpa` now behaves differently. A couple of places it's being referenced are: 

Does `current_wdpa` still return the same value as it previously did? We just need to make sure that we aren't calling it erroneously

Reply:
current_wdpa still returns a bucket, but it will be a different bucket in one case, specifically when the bucket specified by the date constants is not the most recent bucket. unless there is a circumstance where we want the most recent recent bucket to be used even if it hasn't been released yet, i think this is the behaviour we want.

I share your concern about side effects, most of the tests for this class are broken